### PR TITLE
NMS-13358: Move Resource Metadata Handling to TS

### PR DESCRIPTION
**Pull request by patrick-schweizer**
This PR:

- modifies the time series integration layer to writes and read string attributes to the time series database (instead of storing them internally in postgres)
- it removes the usage of the local postgres db for timeseries data.

Open tasks to make it productive:
* remove table itself. Do we need to migrate data?
* remove deprecated methods from ResourceStorageDao and adopt clients of those methods.

### CODE REVIEW

File: features/timeseries/src/main/java/org/opennms/netmgt/timeseries/resource/TimeseriesResourceStorageDao.java [hunk 1]
```diff
@@ -197,6 +216,7 @@ public boolean delete(ResourcePath path) {
     }
 
     @Override
+    // TODO: Patrick: We can remove this method from interface and force clients to use KV store?
```
Comment (patrick-schweizer) [inline features/timeseries/src/main/java/org/opennms/netmgt/timeseries/resource/TimeseriesResourceStorageDao.java:219]: @j-white did I understand that right?

File: features/timeseries/src/main/java/org/opennms/netmgt/timeseries/resource/TimeseriesResourceStorageDao.java [hunk 2]
```diff
@@ -211,16 +231,19 @@ public void setStringAttribute(ResourcePath path, String key, String value) {
...
```
Comment (patrick-schweizer) [inline features/timeseries/src/main/java/org/opennms/netmgt/timeseries/resource/TimeseriesResourceStorageDao.java:234]: @j-white did I understand that right?

File: features/timeseries/src/main/java/org/opennms/netmgt/timeseries/resource/TimeseriesResourceStorageDao.java [hunk 3]
```diff
@@ -211,16 +231,19 @@ public void setStringAttribute(ResourcePath path, String key, String value) {
...
     public String getStringAttribute(ResourcePath path, String key) {
         return getStringAttributes(path).get(key);
     }
 
     @Override
+    // TODO: Patrick: We can remove this method from interface and force clients to use KV store?
```
Comment (patrick-schweizer) [inline features/timeseries/src/main/java/org/opennms/netmgt/timeseries/resource/TimeseriesResourceStorageDao.java:240]: @j-white did I understand that right?

File: features/timeseries/src/main/java/org/opennms/netmgt/timeseries/resource/TimeseriesResourceStorageDao.java [hunk 4]
```diff
@@ -211,16 +231,19 @@ public void setStringAttribute(ResourcePath path, String key, String value) {
...
     public Map<String, String> getStringAttributes(ResourcePath path) {
         return getMetaData(path);
     }
 
     @Override
+    // TODO: Patrick: We can remove this method from interface and force clients to use KV store?
```
Comment (patrick-schweizer) [inline features/timeseries/src/main/java/org/opennms/netmgt/timeseries/resource/TimeseriesResourceStorageDao.java:246]: @j-white did I understand that right?

**Comment by opennms-bamboo** [comment]
Welcome CI/CD System (cicd-system@opennms.com).

We would like to thank you for your contribution.

I could not find a signed OpenNMS Contributor Agreement (OCA) assigned to the username CI/CD System (cicd-system@opennms.com).
In order to maintain some order with respect to the copyright to the code,
every contributor is asked to sign a "Contributor Agreement".

The OpenNMS Contributor Agreement can be downloaded [here](https://wiki.opennms.org/wiki/images/6/6b/ContributorAgreement.pdf).
Please send a signed copy to oca@opennms.org.

If you need more information, please visit https://wiki.opennms.org/wiki/Contributor_Agreement

**Comment by opennms-bamboo** [comment]
Welcome Jian Yet Lee (jylee@opennms.com).

We would like to thank you for your contribution.

I could not find a signed OpenNMS Contributor Agreement (OCA) assigned to the username Jian Yet Lee (jylee@opennms.com).
In order to maintain some order with respect to the copyright to the code,
every contributor is asked to sign a "Contributor Agreement".

The OpenNMS Contributor Agreement can be downloaded [here](https://wiki.opennms.org/wiki/images/6/6b/ContributorAgreement.pdf).
Please send a signed copy to oca@opennms.org.

If you need more information, please visit https://wiki.opennms.org/wiki/Contributor_Agreement

**Comment by j-white** [comment]
> remove table itself. Do we need to migrate data?

Let's drop the tables - no need to worry about migration.

> remove deprecated methods from ResourceStorageDao and adopt clients of those methods.

We'll need to address this before the patch get's merged to `develop`, but we can ignore it for now.

File: opennms-dao-api/src/main/java/org/opennms/netmgt/dao/api/ResourceStorageDao.java [hunk 5]
```diff
@@ -148,6 +154,8 @@
 
     /**
      * Retrieves the meta-data stored at the given path.
+     * Deprecated: if needed store collector specific string attributes in a KV store
      */
+    @Deprecated
```
Comment (j-white) [inline opennms-dao-api/src/main/java/org/opennms/netmgt/dao/api/ResourceStorageDao.java:159]: We can still support and expose this.

File: opennms-dao-api/src/main/java/org/opennms/netmgt/dao/api/ResourceStorageDao.java [hunk 6]
```diff
@@ -117,17 +117,23 @@
...
     public String getStringAttribute(ResourcePath path, String key);
 
     /**
      * Returns the value for the given (resource-level) attribute, or null if it does not exist.
+     * Deprecated: if needed store collector specific string attributes in a KV store
      */
+    @Deprecated
```
Comment (j-white) [inline opennms-dao-api/src/main/java/org/opennms/netmgt/dao/api/ResourceStorageDao.java:136]: We can still support and expose this.

File: opennms-dao-api/src/main/java/org/opennms/netmgt/dao/api/ResourceStorageDao.java [hunk 7]
```diff
@@ -117,17 +117,23 @@
...
     public void setStringAttribute(ResourcePath path, String key, String value);
 
     /**
      * Returns the value for the given (resource-level) attribute, or null if it does not exist.
+     * Deprecated: if needed store collector specific string attributes in a KV store
      */
+    @Deprecated
```
Comment (j-white) [inline opennms-dao-api/src/main/java/org/opennms/netmgt/dao/api/ResourceStorageDao.java:129]: We can still support and expose this.

File: opennms-dao-api/src/main/java/org/opennms/netmgt/dao/api/ResourceStorageDao.java [hunk 8]
```diff
@@ -117,17 +117,23 @@
 
     /**
      * Sets the specified (resource-level) attribute at the given path.
+     * Deprecated: if needed store collector specific string attributes in a KV store
      */
+    @Deprecated
```
Comment (j-white) [inline opennms-dao-api/src/main/java/org/opennms/netmgt/dao/api/ResourceStorageDao.java:122]: This is the only method we need to deprecate - strings should only be set and persisting metrics - but we can still use this interface to retrieve their values.

**Review by j-white — APPROVED** [review]

**Pull request merged**